src/neuroconv/datainterfaces/behavior/fictrac/fictracdatainterface.py

```diff
@@ -102,17 +106,25 @@ def add_to_nwbfile(
 
         import pandas as pd
 
-        fictrac_data_df = pd.read_csv(self.file_path, sep=",", header=None, names=self.data_columns)
+        # The first row only contains the session start time and invalid data
+        fictrac_data_df = pd.read_csv(self.file_path, sep=",", skiprows=1, header=None, names=self.data_columns)
 
         # Get the timestamps
         timestamps_milliseconds = fictrac_data_df["timestamp"].values
         timestamps = timestamps_milliseconds / 1000.0
-        rate = calculate_regular_series_rate(series=timestamps)  # Returns None if it is not regular
-        write_timestamps = True
-        if rate:
-            write_timestamps = False
 
-        processing_module = get_module(nwbfile=nwbfile, name="Behavior")
+        # Shift to the session start time
+        metadata = metadata or self.get_metadata()
+        interface_session_start_time = extract_session_start_time(self.file_path)
+        conversion_session_start_time = metadata["NWBFile"]["session_start_time"]
+        starting_time = interface_session_start_time.timestamp() - conversion_session_start_time.timestamp()
+        timestamps_shifted = timestamps + starting_time
+
+        # Note: The last values of the timestamps look very irregular for the sample file in catalyst neuro gin repo
+        # The reason, most likely, is that FicTrac is relying on OpenCV to get the timestamps from the video
+        # And we know that OpenCV is not very accurate with the timestamps, specially the last ones.
```

Comment:
Remind me why we know that? Past investigations to OpenCV, depending on codec, largely relate to the timestamps being calculated deterministically by `index / fps` equation

Reply:
You mentioned that before, where are you getting that?

It uses the `picture_pst` variable:
https://github.com/opencv/opencv/blob/2f63c583801cb9081cc696e9842fc9a43516904c/modules/videoio/src/cap_ffmpeg_impl.hpp#L1771-L1776

How is this obtained? Well, it tries to get the real presentation timestamp (or the package presentation time) depending on the version:

https://github.com/opencv/opencv/blob/2f63c583801cb9081cc696e9842fc9a43516904c/modules/videoio/src/cap_ffmpeg_impl.hpp#L1575-L1583

The problem is that the fallback when it does not find that is the decoded timestamps which for some reason fails at the end in my experience. This is what was happening on that old issue with Deep Lab Cut where the guy from datalad decided to interpolate the times.

Reply:
Yeah, it is very annoying that this is soo unclear. I guess video people don't really care about the time to the degree that we do and then that explains it.